td-shim-tools/src/tee_info_hash.rs

```diff
@@ -152,12 +172,71 @@ impl TdInfoStruct {
     }
 
     pub fn build_mrtd(&mut self, raw_image_file: &mut File, image_size: u64) {
+        let mut metadata_off: u32 = 0;
+
         raw_image_file
-            .seek(SeekFrom::Start(image_size - TDVF_DESCRIPTOR_OFFSET as u64))
+            .seek(SeekFrom::Start(
+                image_size - OVMF_TABLE_FOOTER_GUID_OFFSET as u64,
+            ))
             .unwrap();
 
-        let metadata_off = raw_image_file.read_u32::<LittleEndian>().unwrap()
-            - size_of::<TdxMetadataGuid>() as u32;
+        let mut footer_guid_buf = [0; size_of::<Guid>()];
+        raw_image_file.read_exact(&mut footer_guid_buf).unwrap();
```

Comment:
- This method contains a lot of `unwrap()` calls, so you might consider making it return a `Result<()>`?
- This method doesn't appear to ever get called...?
- It would be great if you could add a unit test for this method.

Reply:
Thanks! I will follow your comments later to refactor the code.